app/javascript/packages/document-capture/steps/getSteps.ts

```diff
@@ -0,0 +1,95 @@
+import type { FormStep } from '@18f/identity-form-steps';
+import { useI18n } from '@18f/identity-react-i18n';
+import { UploadFormEntriesError } from '../services/upload';
+import DocumentsStep from '../components/documents-step';
+import InPersonPrepareStep from '../components/in-person-prepare-step';
+import InPersonLocationPostOfficeSearchStep from '../components/in-person-location-post-office-search-step';
+import InPersonLocationFullAddressEntryPostOfficeSearchStep from '../components/in-person-location-full-address-entry-post-office-search-step';
+import InPersonSwitchBackStep from '../components/in-person-switch-back-step';
+import ReviewIssuesStep from '../components/review-issues-step';
+import withProps from '../higher-order/with-props';
+
+const getReviewStep = (submissionError) => {
+  const { t } = useI18n();
+
+  return {
+    name: 'review',
+    form:
+      submissionError instanceof UploadFormEntriesError
+        ? withProps({
+            remainingAttempts: submissionError.remainingAttempts,
+            isFailedResult: submissionError.isFailedResult,
+            isFailedDocType: submissionError.isFailedDocType,
+            captureHints: submissionError.hints,
+            pii: submissionError.pii,
+            failedImageFingerprints: submissionError.failed_image_fingerprints,
+          })(ReviewIssuesStep)
+        : ReviewIssuesStep,
+    title: t('errors.doc_auth.rate_limited_heading'),
+  };
+};
+
+export const getSteps = (
+  submissionError,
+  inPersonURL,
+  inPersonFullAddressEntryEnabled,
+  flowPath,
+) => {
+  const { t } = useI18n();
```

Comment:
Is this meant to be a hook?

* If yes: I think we should...
   * ... name it prefixed with `use`
   * ... ensure it's called closer to the top of the render logic in `DocumentCapture`
   * ... consider incorporating more things that can be accessed by hooks vs. passed as parameter (especially context values like `inPersonURL`)
   * ... move this file into `hooks` ?
* If no: We can't use hooks here.

Personally I think this is a good use-case for a hook 👍

Reply:
Thanks for pointing that out. I missed that this is clearly a hook.
- Moved and renamed it ([commit](https://github.com/18F/identity-idp/pull/9433/commits/d6f9adadee1231c3b26bb259a4b644e8f61de755))
- Changed it to use context rather than passing everything in ([commit](https://github.com/18F/identity-idp/pull/9433/commits/acb2eaa7da82373d907a7f4ff64639c7c3d21c84))

Reply:
Next up is tests.

Reply:
Is it possible to use `renderHook` from Testing Library to test the raw hook result rather than a rendered output, like we do in some other specs?

https://github.com/18F/identity-idp/blob/89b930d7cf05e9685abd8d65f76101ea9d1ee508/app/javascript/packages/components/hooks/use-focus-trap.spec.ts#L26

https://testing-library.com/docs/react-testing-library/api/#renderhook (IIRC this was folded into `@testing-library/react` in a newer version than what we use)

Reply:
I'll look at that again, it might make the test better. I stopped investigating when I saw

> You should prefer render since a custom test component results in more readable and robust tests since the thing you want to test is not hidden behind an abstraction.

But I think I ended up hiding it behind a different extraction, so I'll investigate this again.

Reply:
Thanks for the push. I reworked the testing to use `renderHook` ([commit](https://github.com/18F/identity-idp/pull/9433/commits/c95e1401862ee49b08b403a02de7a9c12a940283)) and it feels much better.

I don't love the `createWrapper` code since it requires a lot of thought to understand but I added a comment and I'm pretty convinced it's the best way.